haproxy-ingress/templates/_podtemplate.yaml

```diff
@@ -280,13 +280,23 @@ spec:
 {{- if .Values.controller.enableStaticPorts }}
 - name: http
   containerPort: 80
-  {{- if and (eq .Values.controller.kind "DaemonSet") .Values.controller.daemonset.useHostPort .Values.controller.daemonset.hostPorts.http }}
+  {{- if and (eq .Values.controller.kind "DaemonSet") .Values.controller.daemonset.useHostPort }}
+    {{- if .Values.controller.daemonset.hostIP }}
+  hostIP: {{ .Values.controller.daemonset.hostIP }}
+    {{- end }}
+    {{- if .Values.controller.daemonset.hostPorts.http }}
   hostPort: {{ .Values.controller.daemonset.hostPorts.http }}
+    {{- end }}
   {{- end }}
 - name: https
   containerPort: 443
-  {{- if and (eq .Values.controller.kind "DaemonSet") .Values.controller.daemonset.useHostPort .Values.controller.daemonset.hostPorts.https }}
+  {{- if and (eq .Values.controller.kind "DaemonSet") .Values.controller.daemonset.useHostPort }}
+    {{- if .Values.controller.daemonset.hostIP }}
+  hostIP: {{ .Values.controller.daemonset.hostIP }}
```

Comment:
You're already checking if hostIP was declared before use it, what if use `useHostPort` to only check hostPort? Otherwise this might be confusing for users.

Reply:
You're right, fixed.